pkg/cmd/list.go

```diff
@@ -119,67 +129,67 @@ func NewWhoCanOptions(configFlags *clioptions.ConfigFlags,
 }
 
 func NewCmdWhoCan(streams clioptions.IOStreams) (*cobra.Command, error) {
-	configFlags := clioptions.NewConfigFlags(true)
-
-	clientConfig, err := configFlags.ToRESTConfig()
-	if err != nil {
-		return nil, fmt.Errorf("getting config: %v", err)
-	}
-
-	client, err := kubernetes.NewForConfig(clientConfig)
-	if err != nil {
-		return nil, fmt.Errorf("creating client: %v", err)
-	}
-
-	mapper, err := configFlags.ToRESTMapper()
-	if err != nil {
-		return nil, fmt.Errorf("getting mapper: %v", err)
-	}
-
-	clientNamespace := client.CoreV1().Namespaces()
-	accessChecker := NewAccessChecker(client.AuthorizationV1().SelfSubjectAccessReviews())
-	namespaceValidator := NewNamespaceValidator(clientNamespace)
-	resourceResolver := NewResourceResolver(client.Discovery(), mapper)
-
-	o := NewWhoCanOptions(configFlags,
-		configFlags.ToRawKubeConfigLoader(),
-		clientNamespace,
-		client.RbacV1(),
-		namespaceValidator,
-		resourceResolver,
-		accessChecker,
-		NewPolicyRuleMatcher(),
-		streams)
+	o := newWhoCanOptions()
 
 	cmd := &cobra.Command{
 		Use:          whoCanUsage,
 		Long:         whoCanLong,
 		Example:      whoCanExample,
 		SilenceUsage: true,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if err := o.Complete(args); err != nil {
+			clientConfig, err := o.configFlags.ToRESTConfig()
+			if err != nil {
+				return fmt.Errorf("getting config: %v", err)
+			}
+
+			client, err := kubernetes.NewForConfig(clientConfig)
+			if err != nil {
+				return fmt.Errorf("creating client: %v", err)
+			}
+
+			mapper, err := o.configFlags.ToRESTMapper()
+			if err != nil {
+				return fmt.Errorf("getting mapper: %v", err)
+			}
+
+			clientNamespace := client.CoreV1().Namespaces()
+			accessChecker := NewAccessChecker(client.AuthorizationV1().SelfSubjectAccessReviews())
+			namespaceValidator := NewNamespaceValidator(clientNamespace)
+			resourceResolver := NewResourceResolver(client.Discovery(), mapper)
+
+			wc := NewWhoCan(o,
+				o.configFlags.ToRawKubeConfigLoader(),
+				clientNamespace,
+				client.RbacV1(),
+				namespaceValidator,
+				resourceResolver,
+				accessChecker,
+				NewPolicyRuleMatcher(),
+				streams)
+
+			if err := wc.Complete(args); err != nil {
 				return err
 			}
-			if err := o.Validate(); err != nil {
+			if err := wc.Validate(); err != nil {
 				return err
 			}
-			if err := o.Check(); err != nil {
+			if err := wc.Check(); err != nil {
 				return err
 			}
 
 			return nil
 		},
 	}
 
-	cmd.PersistentFlags().StringVar(&o.subResource, "subresource", o.subResource,
+	cmd.Flags().StringVar(&o.subResource, "subresource", o.subResource,
```

Comment:
I suppose because we don't have any subcommands it doesn't really matter whether these flags are persistent or not?

Reply:
Yes. That was the reason for that change.